### ISSUE TITLE
JSX element class does not support attributes because it does not have a 'props' property.

### ISSUE THREAD

**Issue by soemarko** [react-native-masked-view/masked-view#190]
<img width="675" alt="Screenshot 2023-02-24 at 6 11 42 pm" src="https://user-images.githubusercontent.com/381585/221152680-c6fae2c1-ff86-45cc-915c-8a11510abfdd.png">

Getting that error on vscode, although everything seems to work just fine.

**Comment by bufgix** [comment]
Im getting same error after upgrading expo 48

**Comment by magrinj** [comment]
@soemarko Seems to be fixed on `main` branch, waiting for this fix to be released.
In the meantime you can use `patch-package` with the following patch file:

```patch
diff --git a/node_modules/@react-native-masked-view/masked-view/types/index.d.ts b/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
index acafcc2..6a66a1e 100644
--- a/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
+++ b/node_modules/@react-native-masked-view/masked-view/types/index.d.ts
@@ -3,6 +3,8 @@
 import * as React from 'react';
 import * as ReactNative from 'react-native';
 
+type Constructor<T> = new (...args: any[]) => T;
+
 interface MaskedViewProps extends ReactNative.ViewProps {
   maskElement: React.ReactElement;
   androidRenderingMode?: 'software' | 'hardware';
@@ -11,7 +13,7 @@ interface MaskedViewProps extends ReactNative.ViewProps {
  * @see https://github.com/react-native-masked-view/masked-view
  */
 declare class MaskedViewComponent extends React.Component<MaskedViewProps> {}
-declare const MaskedViewBase: ReactNative.Constructor<
+declare const MaskedViewBase: Constructor<
   ReactNative.NativeMethods
 > &
   typeof MaskedViewComponent;
```

**Comment by nonam4** [comment]
same issue

**Comment by alexi-stein** [comment]
@Naturalclar do you know when the fix will be released?

**Comment by peteygao** [comment]
While waiting for the release, in the mean time you can just use the latest commit in `main` (at the time of this writing) directly via the terminal (syntax for Yarn 2+; Yarn 1 and npm users, you're on your own to figure out the correct syntax, but ironically it's much simpler than Yarn 2+ 😂):
```
yarn add \@react-native-masked-view/masked-view@git+https://github.com/react-native-masked-view/masked-view.git#8b4b2f970e4f0dc3639dd93db8a87fefe286c42e
```

You can also replace the hash with `main` if you want to peg it to the `main` _branch_, but I personally would **_never_** recommend using a non-hash and following a branch; that'll introduce weird breakages in your build down the line when.

**Comment by Naturalclar** [comment]
@alexi-stein thanks for the ping! Just released version v0.2.9 with the fix :rocket:
